addon/plugins/variable-plugin/utils/recreate-variable-uris.ts

```diff
@@ -0,0 +1,43 @@
+import { EXT } from '@lblod/ember-rdfa-editor-lblod-plugins/utils/constants';
+import { sayDataFactory } from '@lblod/ember-rdfa-editor/core/say-data-factory';
+import { OutgoingTriple } from '@lblod/ember-rdfa-editor/core/rdfa-processor';
+import { v4 as uuidv4 } from 'uuid';
+import { Attrs } from '@lblod/ember-rdfa-editor';
+
+export function recreateVariableUris(nodeAttrs: Attrs) {
+  const attrs = { ...nodeAttrs };
+  if (
+    attrs.subject &&
+    (attrs.subject as string).includes(
+      'http://data.lblod.info/mappings/--ref-uuid4-',
+    )
+  ) {
+    attrs.subject = `http://data.lblod.info/mappings/--ref-uuid4-${uuidv4()}`;
+  } else if (
+    attrs.subject &&
+    (attrs.subject as string).includes('http://data.lblod.info/mappings/')
+  ) {
+    attrs.subject = `http://data.lblod.info/mappings/${uuidv4()}`;
+  }
+  attrs.properties = (attrs.properties as OutgoingTriple[]).map((prop) => {
```

Comment:
check out getOutgoingTriple and related utils in `utils/namespace.ts`

Reply:
done :)